include/swift/RemoteInspection/ReflectionContext.h

```diff
@@ -601,7 +601,7 @@ class ReflectionContext
     auto StrTab = reinterpret_cast<const char *>(StrTabBuf);
     bool Error = false;
     auto findELFSectionByName =
-        [&](llvm::StringRef Name) -> std::pair<RemoteRef<void>, uint64_t> {
+        [&](llvm::StringRef Name, bool Retained) -> std::pair<RemoteRef<void>, uint64_t> {
```

Comment:
_Personally_ I think putting a default on a parameter of a lambda doesn't result in the most readable code ever — I think I'd rather write it out explicitly (which I'll have to do in the non-default case anyway).

As for matching, yes, the matching makes sense. The way this works is that there are a couple of compiler flags that enable the stripping of reflection metadata; those work by causing the compiler to _not_ mark the sections as used (i.e. not set the `SHF_GNU_RETAIN` flag). When the compiler builds an object file, it will create a set of reflection metadata sections, all with the same setting of the retain flag. In a normal build, I would expect to find _either_ all retained or all non-retained sections, _except_ where someone was using static libraries (which might be built with a different compiler flag setting). It would be unusual, I think, to vary the flag per object within a build.

If it helps clarify things slightly: the _runtime_ only cares about sections marked as retained, since those are the ones you've told the compiler you want to keep. That's why we only declare the retained ones in `swiftrt.o`. The code in RemoteInspection, by contrast, is interested in seeing what it can find out from the ELF binary, and in that case it _does_ make sense to look for both sets of sections.

Reply:
I've added a large comment to ReflectionContext to explain all of this.